LiteCore/Query/QueryParser.cc

```diff
@@ -1559,34 +1561,39 @@ namespace litecore {
             return;
         }
 
-        // Check out the case the property starts with the result alias.
-        auto resultAliasIter = _aliases.end();
-        if (!property.empty()) {
-            resultAliasIter = _aliases.find(property[0].keyStr().asString());
-            if (resultAliasIter != _aliases.end() && resultAliasIter->second.type != kResultAlias) {
-                resultAliasIter = _aliases.end();
+        // CBL-3040. We should not apply the following rule of result alias if the
+        // property starts with a database collection alias explicitly. In this case,
+        // the following name is the proerty name in the collection.
+        if (!property_starts_with_explicit_alias) {
```

Comment:
I think I probably don't understand the problem. 

What will be the behavior of the following giving that the doc has `c1` and `label1` property?

`SELECT c1 as label1, label1 FROM _`

Reply:
OK, I think I understand it now. The above SQL will not valid. To make it valid, it needs the datasource such as 

`SELECT c1 as label1, mydb.label1 FROM _ AS mydb`

Reply:
Right

Reply:
Before this PR, the above will be parsed as if "SELECT c1 as label1, label1 FROM _" and rejected by Sqlite.

Reply:
Very clear now. Thanks!